src/Pin/Pin.php

```diff
@@ -69,7 +76,12 @@ public function unexport()
     protected function setDirection($direction)
     {
         $directionFile = $this->getPinFile(self::GPIO_PIN_FILE_DIRECTION);
-        $this->fileSystem->putContents($directionFile, $direction);
+        try {
+            $this->fileSystem->putContents($directionFile, $direction);
+        } catch (\RuntimeException $ex) {
+            usleep(100000);
+            $this->fileSystem->putContents($directionFile, $direction);
+        }
```

Comment:
Just wondering if there's a neater way of doing this. I like the idea of retrying a fail - can you think of a way of retrying such that we don't retry for any RuntimeException?

On one hand, if it's a permission denied error there's no point retrying at all. On the other hand, if it's a timing issue it makes sense to maybe even try again sooner (more often).